scripts/powershell/Verify-And-Merge-PRs.ps1

```diff
@@ -0,0 +1,145 @@
+param(
+    [Parameter(Mandatory = $true)]
+    $AuthToken,
+
+    [Parameter(Mandatory = $true)]
+    $PRDataArtifactPath
+)
+
+$PRObjs = [System.Collections.ArrayList]::new()
+$MergedPRs = [System.Collections.ArrayList]::new()
+$ClosedPRs = [System.Collections.ArrayList]::new()
+$UnexpectedState = [System.Collections.ArrayList]::new()
+$OpenAndCleanPRs = [System.Collections.ArrayList]::new()
+$OpenButDirtyPRs = [System.Collections.ArrayList]::new()
+
+
+$GitHubAPIUrlBase = "https://api.github.com/repos/"
+$Headers = @{
+    Authorization = "bearer $AuthToken"
+}
+
+function Merge-PRs()
+{
+    if (($OpenButDirtyPRs.Count -gt 0) -or ($ClosedPRs.Count -gt 0) -or ($UnexpectedState.Count -gt 0))
+    {
+        Write-Output "The following PRs are either closed, or in an Un-mergeable state"
+        Write-Output "If Closed, Please investigate why a Sync PR was closed prematurely."
+        Write-Output "Ensure all checks are green and pr is approved. Merge them manually or run the sync pipeline again"
+
+        foreach ($obj in $OpenButDirtyPRs)
+        {
+            Write-Output "`thttps://github.com/$($obj.RepoOwner)/$($obj.RepoName)/pull/$($obj.PRNumber): State: Un-Mergable"
+        }
+
+        foreach ($obj in $ClosedPRs)
+        {
+            Write-Output "`thttps://github.com/$($obj.RepoOwner)/$($obj.RepoName)/pull/$($obj.PRNumber): State: Closed"
+        }
+
+        foreach ($obj in $UnexpectedState)
+        {
+            Write-Output "`thttps://github.com/$($obj.RepoOwner)/$($obj.RepoName)/pull/$($obj.PRNumber): State: Unexpected"
+        }
+        exit 1
```

Comment:
When in this state to avoid the "powershell exited with 1" dummy error we should use the DevOps error logging method  (see https://github.com/Azure/azure-sdk-tools/blob/master/eng/common/scripts/Verify-Links.ps1#L65) so that the error shows up nicely.

Reply:
I should just move those function into the common.ps1 files that way it can be reused

Reply:
Yes you can put it into common.ps1 later but for now just copy it in this script.